src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt

```diff
@@ -0,0 +1,198 @@
+                tags = emptySet(),
+                author = null,
+                state = null,
+                source = source,
+                isNsfw = false,
+            )
+        }
+
+        override suspend fun getDetails(manga: Manga): Manga {
+            val doc = webClient.httpGet(manga.url.toAbsoluteUrl(domain)).parseHtml()
+            return manga.copy(
+                description = doc.selectFirst("div#syn-target")?.text(),
+                largeCoverUrl = doc.selectFirst(".a1 > figure img")?.src(),
+                tags = doc.select(".a2 div > a[rel='tag'].label").mapNotNullToSet { a ->
+                    MangaTag(
+                        key = a.attr("href").substringAfterLast("/"),
+                        title = a.text().trim(),
+                        source = source,
+                    )
+                },
+                author = doc.selectFirst("div.y6x11p i.fas.fa-user + span.dt")?.text()?.takeUnless {
+                    it.equals("updating", true)
+                },
+                state = when (doc.selectFirst("div.y6x11p i.fas.fa-rss + span.dt")?.text()?.lowercase()) {
+                    "ongoing", "đang tiến hành", "進行中" -> MangaState.ONGOING
+                    "completed", "hoàn thành", "完了" -> MangaState.FINISHED
+                    else -> null
+                },
+                chapters = doc.select("ul > li.chapter").mapChapters { i, element ->
+                    val href = element.selectFirst("a")?.attrAsRelativeUrl("href") ?: element.parseFailed("Chapter link not found")
+                    MangaChapter(
+                        id = generateUid(href),
+                        name = element.selectFirst("a")?.text().orEmpty(),
+                        number = i + 1f,
+                        url = href,
+                        scanlator = null,
+                        uploadDate = element.selectFirst("time[datetime]")?.attr("datetime")?.toLongOrNull()?.times(1000) ?: 0L,
+                        branch = null,
+                        source = source,
+                        volume = 0
+                    )
+                }
+            )
+        }
+
+        override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+            val fullUrl = chapter.url.toAbsoluteUrl(domain)
+            val doc = webClient.httpGet(fullUrl).parseHtml()
+            val script = doc.selectFirst("script:containsData(const CHAPTER_ID)")?.data()
+                ?: throw Exception("Failed to get chapter id")
+
+            val chapterId = script.substringAfter("const CHAPTER_ID = ").substringBefore(";")
+
+            val ajaxUrl = buildString {
+                append("https://")
+                append(domain)
+                append("/ajax/image/list/chap/")
+                append(chapterId)
+            }
+
+            val ajaxResponse = webClient.httpGet(ajaxUrl)
+            val responseJson = JSONObject(ajaxResponse.requireBody().string())
```

Comment:
```suggestion
            val responseJson = webClient.httpGet(ajaxUrl).parseJson()
```
Do not operate with raw response if it is not required